src/utils/idpLoader.ts

```diff
@@ -0,0 +1,48 @@
+import nodeFetch from "node-fetch";
+import { DOMParser } from "xmldom";
+import { IDPEntityDescriptor } from "../types/IDPEntityDescriptor";
+import { log } from "./logger";
+
+const EntityDescriptorTAG = "md:EntityDescriptor";
+const X509CertificateTAG = "ds:X509Certificate";
+const SingleSignOnServiceTAG = "md:SingleSignOnService";
+const SingleLogoutServiceTAG = "md:SingleLogoutService";
+
+export async function parseIdpMetadata(
+  ipdMetadataPage: string
+): Promise<ReadonlyArray<IDPEntityDescriptor>> {
+  const domParser = new DOMParser().parseFromString(ipdMetadataPage);
+  const entityDescriptors = domParser.getElementsByTagName(EntityDescriptorTAG);
+  return Array.from(entityDescriptors).reduce(
+    (idp: ReadonlyArray<IDPEntityDescriptor>, element: Element) => {
+      const certs = Array.from(
+        element.getElementsByTagName(X509CertificateTAG)
+      ).map(cert => {
+        return cert.textContent;
+      });
+      const elementInfoOrErrors = IDPEntityDescriptor.decode({
+        cert: certs,
+        entityID: element.getAttribute("entityID"),
+        entryPoint: (element
+          .getElementsByTagName(SingleSignOnServiceTAG)
+          .item(0) as Element).getAttribute("Location"),
+        logoutUrl: (element
+          .getElementsByTagName(SingleLogoutServiceTAG)
+          .item(0) as Element).getAttribute("Location")
+      });
+      if (elementInfoOrErrors.isLeft()) {
+        log.info("Invalid md:EntityDescriptor. Skipping ...");
```

Comment:
this should be a `log.warning` - I would also log the decoding errors, you can use `ReadableReporter` (search within this project for examples)

Reply:
fixed on commit e281711